src/main/core/manager.c

```diff
@@ -473,23 +474,28 @@ static gchar** _manager_generateEnvv(Manager* manager, InterposeMethod interpose
      *   - preload values from LD_PRELOAD entries in the environment process option */
     GPtrArray* ldPreloadArray = g_ptr_array_new();
 
-    if (manager->preloadShimPath != NULL) {
-        debug("adding Shadow preload shim path %s", manager->preloadShimPath);
-        g_ptr_array_add(ldPreloadArray, g_strdup(manager->preloadShimPath));
-    }
+    // Track if we need to preload the dummy lib.
+    bool need_dummy_preload = true;
```

Comment:
Is there any harm in always preloading the dummy lib?

Reply:
We decided to always preload the dummy lib, and I changed to doing that now in 8cf7d9f.